githooks/pre-commit

```diff
@@ -0,0 +1,44 @@
+#! /bin/bash
+#
+# A pre-commit Git hook.
+#
+# Currently checks the formatting of all Rust files staged for commit,
+# failing if any need to be reformatted using `rustfmt`.  Note that the
+# hook does not modify any file.
+#
+# Authors
+#
+# The Veracruz Development Team.
+#
+# Copyright and licensing
+#
+# See the `LICENSE_MIT.markdown` file in the Veracruz repository root
+# directory for copyright and licensing information.
+
+HAS_ISSUES=0
+FIRST_FILE=1
+
+echo "Checking code formatting of files staged for commit."
+
+for file in $(git diff --name-only --staged *.rs); do
+    RUSTFMT="$(rustfmt --check --skip-children --unstable-features $file)"
+    if [ "$RUSTFMT" != "" ]; then
+        if [ $FIRST_FILE -eq 0 ]; then
+            echo -n ", "
+        fi  
+        printf "[❌]: $file\n"
```

Comment:
Unfortunately ❌ isn't displayed properly with the `xterm` I'm using here, so this could be problematic. Apparently it belongs to the "Dingbats" block. I don't think you necessarily have to stick to ASCII in this day and age but it might be a good idea to go with something slightly less exotic.

Reply:
It looks as though the terminal I'm using here really does only (UTF-8-encoded) ISO-8859-1.

Reply:
I've removed the Unicode entirely in place of an ASCII message, instead.